src/modules/auth/index.ts

```diff
@@ -10,7 +10,8 @@ import LoginResponse from 'types/login-response.interface';
  * @extends {Base}
  */
 export default class Auth extends Base {
-  tokenProvider: TokenProvider;
+  private tokenProvider: TokenProvider;
+  private baseUrl = '/auth';
```

Comment:
@SparoHawk why private and not protected from the base? also where is the type for baseUrl shouldnt it be string?

Reply:
@kaioken TypeScript infers typing for simple object based on the value. It will know by default that `baseUrl` has to be string.

TypeScript is not your typical OOP. If you define a protected property you [HAVE to initialize](https://www.typescriptlang.org/docs/handbook/2/classes.html#protected) it in the constructor. Therefore I'd have to pass the `baseUrl` string during the `super()` call.

Reply:
![image](https://user-images.githubusercontent.com/750563/125553026-289662ae-38d7-45d2-ab3b-94aacd739e8d.png)

This is the message you get when you just declare it.

Reply:
@kaioken I can get around it by doing `protected baseUrl = ''`, and then declaring `readonly baseUrl = '/auth';` in the module.

@smancebo What is your take on this?

Reply:
As baseUrl will be in all the modules, we need to implement the baseUrl field on the Base class as a **protected** member, the **protected** modifier will prevent the baseUrl property being modified on class instances (as the property will not be visible from the instance) but will be able to be modified on extending classes

![image](https://user-images.githubusercontent.com/5271354/125628813-56c1e325-1004-47f8-afa8-3d1cbc00e2e3.png)

![image](https://user-images.githubusercontent.com/5271354/125628864-750e96cb-3a2e-43ea-bd13-decdb6b845df.png)

Reply:
@smancebo VSCode highlights an error when declaring `baseUrl` and not initializing it in the constructor (look at the image I added as an example).

Reply:
@SparoHawk Yes, the reason for this is the linter preventing us to declare an unused variable, setting an initial value let us bypass the linter error and let us assign the property on subclases.